src/java/main/org/apache/zookeeper/server/PrepRequestProcessor.java

```diff
@@ -628,14 +628,7 @@ protected void pRequest(Request request) throws RequestProcessorException {
                 break;
  
             //All the rest don't need to create a Txn - just verify session
-            case OpCode.sync:
-            case OpCode.exists:
-            case OpCode.getData:
-            case OpCode.getACL:
-            case OpCode.getChildren:
-            case OpCode.getChildren2:
-            case OpCode.ping:
-            case OpCode.setWatches:
+            default:
```

Comment:
I get your point. I would argue that we perform the check to make sure we don't get a bad OpCode here: https://github.com/apache/zookeeper/blob/branch-3.4/src/java/main/org/apache/zookeeper/server/ZooKeeperServer.java#L742 and I think this is cleaner. I would be willing to change this if you feel strongly about it.